ci-operator/step-registry/assisted/ofcir/setup/assisted-ofcir-setup-commands.sh

```diff
@@ -0,0 +1,203 @@
+ANSIBLE_CONFIG_FILE="${SHARED_DIR}/ansible.cfg"
+
+if [[ ! -f "$CIRFILE" ]]; then
+    echo "Error: CIR file not found at $CIRFILE"
+    exit 1
+fi
+
+if [[ ! -f "$PACKET_CONF" ]]; then
+    echo "Error: packet-conf.sh not found at $PACKET_CONF"
+    exit 1
+fi
+
+if [[ ! -f "$SSH_KEY_FILE" ]]; then
+    echo "Error: SSH key file not found at $SSH_KEY_FILE"
+    exit 1
+fi
+
+echo "executing packet-conf.sh..."
+# shellcheck disable=SC1090
+source "$PACKET_CONF"
+
+export SSH_KEY_FILE="$SSH_KEY_FILE"
+
+if [[ -z "${IP:-}" ]]; then
+    IP=$(jq -r '.ip' "$CIRFILE")
+fi
+export IP
+
+mkdir -p build/ansible
+cd build/ansible
+
+cat > prepare-ansible.yaml <<-EOF
+- name: Prepare locally
+  hosts: localhost
+  collections:
+    - community.general
+  gather_facts: no
+  vars:
+    ansible_remote_tmp: ../tmp
+    SHARED_DIR: "{{ lookup('env', 'SHARED_DIR') }}"
+  tasks:
+    - name: Check if ansible inventory exists
+      stat:
+        path: "{{ SHARED_DIR }}/inventory"
+      register: inventory
+    - name: Create default ansible inventory
+      ansible.builtin.copy:
+        dest: "{{ SHARED_DIR }}/inventory"
+        content: |
+          [primary]
+          primary-{{ lookup('env', 'IP') }} ansible_host={{ lookup('env', 'IP') }} ansible_user=root ansible_ssh_user=root ansible_ssh_private_key_file={{ lookup('env', 'SSH_KEY_FILE') }} ansible_ssh_common_args="-o ConnectTimeout=5 -o UserKnownHostsFile=/dev/null -o ServerAliveInterval=90 -o LogLevel=ERROR"
+      when: not inventory.stat.exists
+    - name: Create ssh config file
+      ansible.builtin.copy:
+        dest: "{{ SHARED_DIR }}/ssh_config"
+        content: |
+          Host ci_machine
+            User root
+            HostName {{ lookup('env', 'IP') }}
+            ConnectTimeout 5
+            StrictHostKeyChecking no
+            ServerAliveInterval 90
+            LogLevel ERROR
+            IdentityFile {{ lookup('env', 'SSH_KEY_FILE') }}
+            ConnectionAttempts 10
+    - name: Create ansible configuration
+      ansible.builtin.copy:
+        dest: "{{ SHARED_DIR }}/ansible.cfg"
+        content: |
+          [defaults]
+          callback_whitelist = profile_tasks
+          host_key_checking = False
+
+          verbosity = 2
+          stdout_callback = yaml
+          bin_ansible_callbacks = True
+
+          [ssh_connection]
+          retries = 10
+EOF
+
+ansible-playbook prepare-ansible.yaml
+export ANSIBLE_CONFIG="${ANSIBLE_CONFIG_FILE}"
+
+cat > ensure-memory.yaml <<-"EOF"
+- name: Make sure inventory contains at least one host
+  hosts: localhost
+  tasks:
+    - fail:
+        msg: "[ERROR] Empty inventory. No host available."
+      when: groups.all|length == 0
+      
+- name: Ensure minimum total memory (RAM + swap)
+  hosts: all
+  gather_facts: true
+
+  vars:
+    target_gib: "{{ lookup('env', 'REQUIRED_MEMORY_GIB') | int }}"
+    swapfile_basename: swapfile-ci
+
+  tasks:
+    - name: Determine mount with most available space
+      set_fact:
```

Comment:
```suggestion
      ansible.buildtin.set_fact:
```

Using task fqdn (here and for all other tasks in this file) is a best practice: https://ansible.readthedocs.io/projects/lint/rules/fqcn/